hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/connector/AbstractUpdateMapP.java

```diff
@@ -54,6 +55,15 @@
  */
 public abstract class AbstractUpdateMapP<T, K, V> extends AsyncHazelcastWriterP {
 
+    /**
+     * This processor uses {@link IMap#submitToKeys(Set, EntryProcessor)}, which
+     * if used from multiple parallel async operations can end up reordering
+     * the changes done to the map and this in turn can result in unforseen
+     * consequences. For this reason we need to limit ourselves to a single
+     * in-flight operation at a time.
+     */
+    private static final int MAX_PARALLEL_ASYNC_OPS = 1;
```

Comment:
this will decrease performance for existing UpdateMapP, we should only do this for the CDCSink.